### PR TITLE
Expose an option to control the number of Shake threads

### PR DESCRIPTION
**Pull request by pepeiborra**
/cc @mpickering

### CODE REVIEW

File: hie.yaml [hunk 1]
```diff
@@ -1 +1,3 @@
-cradle: {stack: {component: "ghcide:lib"}}
+# cradle: {cabal: {component: "ghcide:exe:ghcide"}}
+# cradle: {cabal: {component: "ghcide:ghcide-tests"}}
+cradle: {cabal: {component: "ghcide"}}
```
Comment (mpickering) [inline hie.yaml]: Accident.

File: test/exe/Main.hs [hunk 2]
```diff
@@ -2094,7 +2094,7 @@ runInDir dir s = do
       -- which can be quite useful for debugging.
       -- { logStdErr = True, logColor = False }
       -- If you really want to, you can also see all messages
-      -- { logMessages = True, logColor = False }
+      { logMessages = True, logColor = True}
```
Comment (mpickering) [inline test/exe/Main.hs]: Accident?

File: hie.yaml (hunk 1 above)
Comment (pepeiborra) [reply hie.yaml]: oops, fixed

File: test/exe/Main.hs (hunk 2 above)
Comment (pepeiborra) [reply test/exe/Main.hs]: fixed too

**Pull request merged**